### PR TITLE
 [Core] Move RStringView.h to ROOT/RStringView.hxx

### PR DESCRIPTION
**Pull request by bluehood**
This is done for consistency with the other backports.
The old header RStringView.h is left for backwards compatibility, and it simply includes ROOT/RStringView.hxx.

Due to our cmake rules this change requires a removal of $ROOTSYS/core
(or anyway a regeneration of the cmake rules for it) in build
directories used for incremental builds.

I think of this PR as a "nice to have", and I see no downsides, but feel free to close it if you are against such a change.

@vgvassilev I did not know how to update `build/unix/module.modulemap` so I left it as is, let me know if it requires changes.

### CODE REVIEW

File: build/unix/module.modulemap [hunk 1]
```diff
@@ -14,7 +14,7 @@ module ROOT_Types {
   // FIXME: This module should contain only header files with types.
   // The listed below headers are here because we want them to work in both
   // -frtti and -fno-rtti environment. Move them into a separate module.
-  module "RStringView.h" { header "RStringView.h" export * }
+  module "RStringView.h" { header "RStringView.h" export * } // TODO substitute with ROOT/RStringView.hxx
```
Comment (vgvassilev) [inline build/unix/module.modulemap]: Putting `header "ROOT/RStringView.cxx"` should suffice.

Comment (bluehood) [reply build/unix/module.modulemap]: ok so I should leave `module "RStringView.h"` as it is?

Comment (vgvassilev) [reply build/unix/module.modulemap]: Well, as you prefer. The module name is just an identifier. Maybe for consistency we should prefix it with ROOT/, too.

Comment (bluehood) [reply build/unix/module.modulemap]: alright thanks, done

**Pull request merged**